### PR TITLE
Add theoretical bounds

### PR DESCRIPTION
**Pull request by gtklocker**

### CODE REVIEW

**Comment by gtklocker** [comment]
@dionyziz ptal

File: construction.tex [hunk 1]
```diff
@@ -138,3 +138,85 @@ \section{Interlinks as sets of superblocks}\label{sec.construction}
...
+\begin{proof}
+  \begin{align*}
+    \EX[B^n]
+    &= -\frac{1}{2} \sum_{i = 1}^n (-1)^i {n \choose i} \frac{2^i}{2^i - 1} \\
+    &= \frac{1}{2}((-\sum_{i=1}^n (-1)^i {n \choose i} \sum_{j=0}^k 2^{-ij}) + (-\sum_{i=1}^n (-1)^i {n \choose i} \frac{1}{2^{(k+1)i} - 2^{ki}})) \\
+    &< \frac{1}{2}((\lg(n) + 2) + e)
+  \end{align*}
+\end{proof}
+
+We obtain that $\EX[B^n] < \frac{1}{2}(\lg(n)+2+e)$. Therefore the expected savings are $\lg(n) - \EX[B^n] > \frac{1}{2}(\lg(n)-2-e)$. Solving for $n$ we obtain that $n > 111$ for savings to exist.
```
Comment (dionyziz) [inline construction.tex]: we obtain that $n > 111$ *is required* for savings to exist

File: construction.tex [hunk 2]
```diff
@@ -138,3 +138,85 @@ \section{Interlinks as sets of superblocks}\label{sec.construction}
...
 \end{figure*}
+
+\begin{lemma}
+  For any integers $i > 0$ and $k \ge 0$,
+  \[
+    \frac{2^i}{2^i-1} = \frac{1}{2^{(k+1)i}-2^{ki}} + \sum_{j=0}^k 2^{-ij}
+  \]
+\end{lemma}
+\begin{proof}
+  We work using induction on $k$.
```
Comment (dionyziz) [inline construction.tex]: Hide induction behind a minimality argument.

File: construction.tex [hunk 3]
```diff
@@ -138,3 +138,85 @@ \section{Interlinks as sets of superblocks}\label{sec.construction}
   \label{fig.log2}
   \end{center}
 \end{figure*}
```
Comment (dionyziz) [inline construction.tex]: Since we have theoretical results below, the above observation and figure can be removed.

File: construction.tex [hunk 4]
```diff
@@ -138,3 +138,85 @@ \section{Interlinks as sets of superblocks}\label{sec.construction}
...
+  For any positive integers $n$ and $k=\lceil \lg(n) \rceil$,
+  \[
+    -\sum_{i=1}^n (-1)^i {n \choose i} \sum_{j=0}^k 2^{-ij} < \lg(n) + 2
+  \]
+\end{lemma}
+\begin{proof}
+  \begin{align*}
+    -\sum_{i=1}^n (-1)^i {n \choose i} \sum_{j=0}^k 2^{-ij}
+    &= k+1 - \sum_{j=0}^k \sum_{i=0}^n (-1)^i {n \choose i} 2^{-ij} \\
+    &= k+1 - \sum_{j=0}^k (1-2^{-j})^n < \lg(n) + 2 \\
```
Comment (dionyziz) [inline construction.tex:169]: I don't follow how we got here from the previous line.

File: construction.tex [hunk 5]
```diff
@@ -138,3 +138,85 @@ \section{Interlinks as sets of superblocks}\label{sec.construction}
...
+  \end{align*}
+\end{proof}
+
+\begin{lemma}
+  For any positive integer $n$ and $k=\lceil \lg(n) \rceil$,
+  \[
+    -\sum_{i=1}^n (-1)^i {n \choose i} \frac{1}{2^{(k+1)i} - 2^{ki}} < e
```
Comment (dionyziz) [inline construction.tex:176]: Shouldn't this be \leq?

File: construction.tex [hunk 6]
```diff
@@ -138,3 +138,85 @@ \section{Interlinks as sets of superblocks}\label{sec.construction}
...
+  \]
+\end{lemma}
+\begin{proof}
+  \begin{align*}
+    -\sum_{i=1}^n (-1)^i {n \choose i} \frac{1}{2^{(k+1)i} - 2^{ki}}
+    &< \sum_{i=1}^n {n \choose i} \frac{1}{2^{(k+1)i} - 2^{ki}} \\
```
Comment (dionyziz) [inline construction.tex]: \leq?

File: construction.tex [hunk 7]
```diff
@@ -138,3 +138,85 @@ \section{Interlinks as sets of superblocks}\label{sec.construction}
...
+    &< \sum_{i=0}^n {n \choose i} \frac{1}{2^{ki}(2^i-1)} \\
```
Comment (dionyziz) [inline construction.tex]: How did we get to i=0?

File: construction.tex [hunk 8]
```diff
@@ -138,3 +138,85 @@ \section{Interlinks as sets of superblocks}\label{sec.construction}
...
+    &< \sum_{i=0}^n {n \choose i} \frac{1}{2^{ki}} \\
+    &\le (1+\frac{1}{n})^n < e \\
```
Comment (dionyziz) [inline construction.tex]: Not sure how we got here from the line above.

**Review by dionyziz — CHANGES_REQUESTED** [review]
Add acknowledgment section at the end and mention Nikos.

File: construction.tex (hunk 5 above)
Comment (gtklocker) [reply construction.tex:176]: no, why?

File: construction.tex (hunk 4 above)
Comment (gtklocker) [reply construction.tex:169]: binomial expansion, explain more clearly in the new version

File: construction.tex (hunk 7 above)
Comment (gtklocker) [reply construction.tex]: isn't the previous amount sum smaller than the sum with an extra positive number added?

Comment (gtklocker) [reply construction.tex]: we want the i=0 to do a binomial expansion afterwards

File: construction.tex (hunk 8 above)
Comment (gtklocker) [reply construction.tex]: added explanation

File: construction.tex (hunk 2 above)
Comment (gtklocker) [reply construction.tex]: is this necessary? do you think it would look much better?

File: construction.tex (hunk 7 above)
Comment (dionyziz) [reply construction.tex]: The denominator is 0.

Comment (gtklocker) [reply construction.tex]: Well you're not wrong

Comment (gtklocker) [reply construction.tex]: Let me see what I can do about it

File: construction.tex (hunk 2 above)
Comment (dionyziz) [reply construction.tex]: If we're going to keep it as forward induction, at least we should shorten:

"By induction on $k$. Clearly it holds for $k = 0$. For $k > 0$ we have:" and the derivation can follow. If you want at some point to say "From the inductive hypothesis" at the point where you do the replacement for $k$ you can add it. I don't think we need to write out the base, the hypothesis, and the step so verbosely.

File: construction.tex (hunk 7 above)
Comment (gtklocker) [reply construction.tex]: @dionyziz ok have a look at the updated version, I think it should be fine now

File: construction.tex (hunk 2 above)
Comment (gtklocker) [reply construction.tex]: Fair, fixing it.

Comment (gtklocker) [reply construction.tex]: done

**Review by dionyziz — APPROVED** [review]

**Pull request merged**